src/server/modules/job/JobController.ts

```diff
@@ -42,17 +43,56 @@ export class JobController {
           await this.jobManagementService.createJobItem({
             params: <JSON>(<unknown>messageParams),
             jobId: job.id,
+            jobItemId: `${job.uuid}/${idx}`,
           })
           await this.sqsService.sendMessage(messageParams)
           return
         }),
       )
       dogstatsd.increment('job.start.success', 1, 1)
-    } catch (e) {
-      logger.error(`error creating and starting job: ${e}`)
+    } catch (error) {
+      logger.error(`error creating and starting job: ${error}`)
       dogstatsd.increment('job.start.failure', 1, 1)
     }
   }
+
+  public updateJobItem: (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) => Promise<void> = async (req, res, next) => {
+    const { jobItemId, status } = req.body
+    try {
+      const jobItem = await this.jobManagementService.updateJobItemStatus(
+        jobItemId,
+        status,
+      )
+      // add jobItem to req.body so that downstream controllers can access it
+      req.body.jobItem = jobItem
+      dogstatsd.increment('jobItem.update.success', 1, 1)
+      res.ok(jsonMessage('successfully updated'))
```

Comment:
> would it be a good idea to merge this method updateJobItem together with updateJob? am thinking:
> - it seems like res.ok() should come only after both the job item and job status have been updated successfully;
> - merging them together avoids this method having to use req.body to store the jobItem for the next one;
> - since the two methods are always executed together and one after the other, it makes sense to put them together?

Actually I kept the two controller methods separate since I was thinking that the job item update response for the Lambda (only responsible for the job item) should not be affected by whether the job was updated successfully or not. This is why I called `next()` to pass the control on, only after responding `res.ok()`. 

With regards to storing `jobItem` in `reqBody`, what could make more sense is storing `jobId` in `reqBody` before passing on the control. This could allow the controller method to be reused in the future by exposing another endpoint, for example _if we wanted to trigger an ad-hoc refresh on the status_ (say a job item was manually updated).

Reply:
Fixed this here, what do you think? [9739842](https://github.com/opengovsg/GoGovSG/pull/2085/commits/973984202893cfb13c20f4ae8df7f25cae3a7c34)

Reply:
i see! makes sense to separate them out if the callback is specifically for job items only. https://github.com/opengovsg/GoGovSG/pull/2085/commits/973984202893cfb13c20f4ae8df7f25cae3a7c34 also looks good